src/backend/providers/streamflix.ts

```diff
@@ -0,0 +1,87 @@
+import { proxiedFetch } from "@/backend/helpers/fetch";
+import { registerProvider } from "@/backend/helpers/register";
+import {
+  MWCaptionType,
+  MWStreamQuality,
+  MWStreamType,
+} from "@/backend/helpers/streams";
+import { MWMediaType } from "@/backend/metadata/types";
+
+const streamflixBase = "https://us-west2-compute-proxied.streamflix.one";
+
+const qualityMap: Record<number, MWStreamQuality> = {
+  360: MWStreamQuality.Q360P,
+  540: MWStreamQuality.Q540P,
+  480: MWStreamQuality.Q480P,
+  720: MWStreamQuality.Q720P,
+  1080: MWStreamQuality.Q1080P,
+};
+
+registerProvider({
+  id: "streamflix",
+  displayName: "StreamFlix",
+  disabled: false,
+  rank: 69,
+  type: [MWMediaType.MOVIE, MWMediaType.SERIES],
+
+  async scrape({ media, episode, progress }) {
+    if (!this.type.includes(media.meta.type)) {
+      throw new Error("Unsupported type");
+    }
+
+    progress(30);
+    if (media.meta.type === MWMediaType.SERIES) {
+      const seasonNumber = media.meta.seasonData.number;
+      const episodeNumber = media.meta.seasonData.episodes.find(
+        (e) => e.id === episode
+      )?.number;
+
+      const streamRes = await proxiedFetch<any>(
+        `/api/player/tv?id=${media.tmdbId}&s=${seasonNumber}&e=${episodeNumber}`,
+        {
+          baseURL: streamflixBase,
+        }
```

Comment:
```suggestion
      const streamRes = await proxiedFetch<any>(
        `/api/player/tv`,
        {
          baseURL: streamflixBase,
          params: {
              id: media.tmdbId,
              s: seasonNumber,
              e: episodeNumber
          },
        }
```

You can also just set the seasonNumber and episodeNumber to 1 if its a movie, or keep it undefined.
Then you dont need such a huge if/else with basically the same code repeating